static/js/components/google_map.tsx

```diff
@@ -0,0 +1,257 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for embedding a map from Google Maps
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import React from "react";
+
+const DEFAULT_MAP_ZOOM = 4;
+
+// number of digits after decimal point in LatLong/<lat>_<long> dcids
+const LAT_LONG_PRECISION = 5;
+
+interface GoogleMapPropType {
+  /**
+   * DCID of the place/event to show a map for.
+   */
+  dcid: string;
+}
+
+interface GoogleMapStateType {
+  /**
+   * Stores responses from api calls.
+   */
```

Comment:
since this refers to the entire type (and not the first prop), goes above the interface.

Reply:
Thanks for the pointer! Updated the comment styling. (This interface got implemented, switched to inline comments about each item)